contrib/completion/bash/sdk

```diff
@@ -76,7 +76,7 @@ __sdkman_complete_candidate_version() {
 			;;
 	esac
 
-	read -r -a COMPREPLY < <(compgen -W "${candidates[*]}" -- "$candidate_version")
+	mapfile -t COMPREPLY < <(compgen -W "${candidates[*]}" -- "$candidate_version")
```

Comment:
`mapfile` requires Bash 4+, so this may introduce a breaking change for some of our users (even if using `mapfile` is technically better what using `read`).

Reply:
The original implementation was using an horrible word splitting on `IFS` using unquoted variable expansion, which also expand globbing patterns within variable. I could manage to have `read -a` loads the newline delimited entries generated by the `compgen` command. That put it at the 15 years old obsolete Bash 3.2 compatibility. Now it will sure not work with Zsh, which use its own custom grammar for arrays.

As for the failing tests. I really can't make sense of it at all.